### PR TITLE
handle duplicate vlan ids correctly (bsc#988050)

### PR DESCRIPTION
**Pull request by gabi2**

### CODE REVIEW

File: src/modules/FcoeClient.rb [hunk 1]
```diff
@@ -500,59 +501,49 @@ def GetFcoeInfo(net_devices)
...
-      ) { |dev| Builtins.foreach(fcoe_info) do |line|
-        # Check whether there is a line for the given interface, e.g.
-        # eth3            | 200  | 00:0d:ec:a2:ef:00\n
-        # Get VLAN channel from second column and FCF MAC from third.
-        line = Builtins.deletechars(line, " \t")
-        columns = Builtins.splitstring(line, "|")
-        if Ops.get(columns, 0, "") == dev
-          # get VLAN and FCF MAC and add it to vlan_info
-          vlan_interface = { "vlan" => Ops.get(columns, 1, ""), "fcf" => Ops.get(columns, 2, "") }
+      net_devices.each { |dev| fcoe_info.each do |line|
```
Comment (teclator) [inline src/modules/FcoeClient.rb]: NP: I prefer a line per loop

File: src/modules/FcoeClient.rb [hunk 2]
```diff
@@ -500,59 +501,49 @@ def GetFcoeInfo(net_devices)
...
-            elsif (vlan_interface["vlan"] || "") == "0" # for VLAN = 0 replace existing entry
-              # VLAN = 0 'wins' (see bnc #813621, comment #4)
-              vlans = Builtins.maplist(vlans) do |vlan|
-                if (vlan["fcf"] || "") == (vlan_interface["fcf"] || "")
-                  Builtins.y2milestone("VLAN = 0 is taken")
-                  Ops.set(vlan, "vlan", "0")
-                end
-                deep_copy(vlan)
+              vlans << vlan_interface
+            end
```
Comment (teclator) [inline src/modules/FcoeClient.rb]: Maybe something like 

```vlans << vlan_interface if vlans.none? {|vlan| vlan["vlan"].to_s == vlan_interface["vlan].to_s }```

File: test/GetVlanInterfaces_spec.rb [hunk 3]
```diff
@@ -8,21 +8,57 @@
...
                                   { "vlan" => "300", "fcf" => "00:0d:ec:a2:ef:01" }],
-                       "eth2" => [{ "vlan" => "200", "fcf" => "00:0d:ec:a2:ef:02" }]
+                       "eth1" => [{ "vlan" => "400", "fcf" => "00:ef:af:34:12:ae" }],
+                       "eth2" => [{ "vlan" => "200", "fcf" => "00:0d:ec:a2:ef:03" }]
                       })
       end
     end
   end
+
+  describe "#GetVlanInterfaces" do
```
Comment (teclator) [inline test/GetVlanInterfaces_spec.rb]: It is not other method that you are describing but other context, so it's not needed.

File: test/GetVlanInterfaces_spec.rb [hunk 4]
```diff
@@ -8,21 +8,57 @@
   end
```
Comment (teclator) [inline test/GetVlanInterfaces_spec.rb]: I think that with
```
before do
  subject.main
end
```

should be enough and replacing ```@fcoe``` by subject

**Comment by teclator** [comment]
NP: Just some small comments about more cleanups that could be out of the scope of the current fix.

Apart of that LGTM

**Comment by gabi2** [comment]
Thanks for review, I will merge it like it is for now.

**Pull request merged**